subsys/bluetooth/audio/pacs.c

```diff
@@ -190,26 +211,62 @@ static ssize_t snk_loc_read(struct bt_conn *conn,
 			    const struct bt_gatt_attr *attr, void *buf,
 			    uint16_t len, uint16_t offset)
 {
+	int err;
+	enum bt_audio_location location;
+	uint32_t location_32;
+
 	BT_DBG("conn %p attr %p buf %p len %u offset %u", conn, attr, buf, len,
 	       offset);
 
-	return bt_gatt_attr_read(conn, attr, buf, len, offset, &snk_loc,
-				 sizeof(snk_loc));
+	err = get_pac_loc(NULL, BT_AUDIO_SINK, &location);
+	if (err != 0) {
+		BT_DBG("get_pac_loc returned %d", err);
+		return BT_GATT_ERR(BT_ATT_ERR_UNLIKELY);
+	}
+
+	location_32 = (uint32_t)location;
+	if (location_32 > BT_AUDIO_LOCATION_MASK || location_32 == 0) {
+		BT_ERR("Invalid location value: 0x%08X", location_32);
+		return BT_GATT_ERR(BT_ATT_ERR_UNLIKELY);
+	}
+
+	return bt_gatt_attr_read(conn, attr, buf, len, offset,
+				 &sys_cpu_to_le32(location_32),
```

Comment:
Doesn't the result have to be assigned to a variable before applying the `&` operator (so that the operand is an "lvalue")?
(Also elsewhere)

Reply:
I tried to use the same thing in some code here, and I get compile errors for both variants.
In my understanding, this should not work - I think one ends up with a value, not a variable, and one can take take the address of a value.
Will check again - I may not have tested correctly.

Reply:
Tested again.  This time it worked for the "le" variant, but fails for the "be" variant.
```
	uint32_t location;
	uint32_t *p;

	location = 7;
	p = &sys_cpu_to_be32(location);
```
gives 
error: lvalue required as unary '&' operand
  324 |  p = &sys_cpu_to_be32(location);
      |      ^

Reply:
Fixed